mutation_writer/multishard_writer.cc

```diff
@@ -131,16 +132,25 @@ future<stop_iteration> multishard_writer::handle_mutation_fragment(mutation_frag
     auto f = make_ready_future<>();
     if (mf.is_partition_start()) {
         _consumed_partitions++;
-        if (unsigned shard = shard_for_mf(mf); shard != _current_shard) {
-            _current_shard = shard;
-            if (!bool(_shard_writers[shard])) {
-                f = make_shard_writer(shard);
+        if (auto shards = shard_for_mf(mf); shards != _current_shards) {
+            _current_shards = shards;
+            for (auto shard : shards) {
+                if (!bool(_shard_writers[shard])) {
```

Comment:
Isn't this loop slow(ish) on very wide nodes?

Reply:
What do you mean? `shards` are only shards for the token, so at most 2. The inner if branch is amortized by tablets size, it's taken at most once per tablet.

Reply:
I somehow misread it as a loop over _shard_writers, and then checking if it's relevant for mf. The loop is fine.